docs/service_x509_credentials.md

```diff
@@ -0,0 +1,101 @@
+Once a domain has been registered in Athenz, the administrator will
+register service identities that are specified in domain roles and
+policy assertions. The latter can reference those roles having access to
+specified resources. Athenz supports service authentication with
+two types of credentials:
+
+- X.509 Certificates (preferred approach)
+- Public/Private Key Pairs [Instructions](reg_service_guide.md)
+
+To contact Athenz Services (ZMS/ZTS) or other Athenz Enabled services,
+the client service must first obtain an Athenz CA issued X.509 certificate.
+In this section we'll cover how to register the service identity and
+obtain X.509 certificates based on what environment your service will be running in.
+
+## Use Case 1: Service Running in AWS EC2
+
+If the service is running in AWS, Service Identity Agent (SIA) running on the
+instance is responsible for generating a private key for the service and
+obtaining a x.509 certificate from ZTS Service. The files are located in
+the following locations:
+
+```
+    private-key: /var/lib/sia/keys/<domain>.<service>.key.pem
+    certificate: /var/lib/sia/certs/<domain>.<service>.cert.pem
+```
+
+The certificate is valid for 30 days and SIA agent automatically
+will refresh the certificate daily. Follow [these steps](service_x509_credentials_aws.md) for
+full details how to configure SIA agent running in AWS as part
+of the foundation image.
+
+## Use Case 2: Service running on-prem within an Athenz Enabled Framework (K8S Omega/Ensemble, SD v3/v4)
```

Comment:
Removed screwdriver and aws-builder-creds setup (which isn't open source). Original text:

 ## Screwdriver Deployment Setup

 Screwdriver platform now supports Athenz X.509 Client certificates for
 all of its projects so you can use the REST API or the aws-builder-creds
 utility that the Athenz team provides to fetch temporary AWS credentials
 for your deployment task. You would still need to follow the complete
 steps defined in this document for your project.

 Within your Screwdriver job, you would fetch and install the aws-builder-creds
 utility:

 ```
 sudo yum-config-manager --add-repo https://edge.artifactory.yahoo.com:4443/artifactory/athens_rpms/athens_rpms.repo
 sudo yum install -y aws-builder-creds
 ```

 Once you have the utility installed, you can retrieve your aws temporary credentials (replace `athenz-domain` and `aws-iam-role-name` with their respective values for your environment):

 In SDv3, use

 ```
 aws-builder-creds -k /tokens/key -c /tokens/cert -d <athenz-domain> -r <aws-iam-role-name> -z zts.athenz.ouroath.com [-t <durationSeconds>] [-e <externalId>] > ~/.aws/credentials
 ```

 In SDv4, use

 ```
 aws-builder-creds -k /sd/tokens/key -c /sd/tokens/cert -d <athenz-domain> -r <aws-iam-role-name> -z zts.athenz.ouroath.com [-t <durationSeconds>] [-e <externalId>] > ~/.aws/credentials
 ```

 Finally, once you create your deployment role in Athenz in step 1
 in the [Athenz AWS Assume Role Configuration Setup](#athenz-aws-assume-role-configuration-setup)
 section, you would add your Screwdriver Project Identity in the
 format `cd.screwdriver.project.sd<project-id>` to that role.

 You might consider using `aws_ansible_membership` SD v3 platform for your project.
 Even though it is developed by the membership team, it is fairly generic and can
 be used to deploy other services to AWS.